transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -337,39 +340,33 @@ public void run() {
         // Register all channels to the new Selector.
         int nChannels = 0;
         for (;;) {
-            try {
-                for (SelectionKey key: oldSelector.keys()) {
-                    Object a = key.attachment();
-                    try {
-                        if (!key.isValid() || key.channel().keyFor(newSelector) != null) {
-                            continue;
-                        }
+            for (SelectionKey key: oldSelector.keys()) {
```

Comment:
I wonder if you should be sync-ing on the `keys()` set - that's what all the JDK classes do and it's iterated in multiple places here? That seems to be the only sane way to do that? I also see that we close Selectors potentially concurrently that causes a `set.clear()` call in the selector and therefore might trigger the CME